docs: revise readme

inventarSarah opened the pull request:
A summary of the changes:
- fix typos and the likes
- add "alerts" to highlight information and make text more interesting
- add list of integrations at the top
- put sponsor/donor-related information into one section
- add a new section that describes the basics of writing tests
  - moved some content from the "best practices" and "integration" sections in there but also added new content
- streamline integrations section to focus on their benefits and what sets them apart (repeating information (eg what types of locations you can provide) have been moved to the "get started writing tests" section
- restructure Best practice section
  - moved "basic" information to the new section
  - created a subsection that talks about location-specific tips
  - moved everything that is not a best practice into a new section, "things to keep in mind" 
  - I also added emojis here - it helps make the text look more interesting and because we want to have more interesting headers here anyway, I thought it would be a good fit
- added a support and feedback section

jimaek commented on README.md:
```diff
@@ -2,54 +2,120 @@
+- [CLI tool](#globalping-cli) for people who feel at home in their terminal
+- [Web tools](#web-based-tools---globalping-website) for on-the-go testing and visual presentations
+- [Slack app](#slack-app) for bringing network testing capabilities into your Slack workspace
+- [GitHub bot](#github-bot) for adding network test results to public GitHub issues
```
Maybe lets add another line at the end with something like "More integrations can be found on our website https://www.jsdelivr.com/globalping/integrations"

inventarSarah replied:
added!

jimaek commented on README.md:
```diff
@@ -2,54 +2,120 @@
+Discover more about Globalping on our website [www.jsdelivr.com/globalping](https://www.jsdelivr.com/globalping)
 
-We thank our sponsors who contribute to the development of Globalping and help us expand the network of probes.
+## Our major donors
```
Donors implies money, but these companies only provide us with probe hosting. So I think "sponsor" works better

inventarSarah replied:
Updated to sponsors

jimaek commented on README.md:
```diff
@@ -2,54 +2,120 @@
+Here are some ways you can help:
+
+- [Run multiple probes](https://github.com/jsdelivr/globalping-probe): Globalping relies on a globally distributed network of probes for running network measurement requests. We're happy to list anyone who can host at least six probes as a donor on GitHub and our website.
+- [Become a GitHub sponsor](https://www.jsdelivr.com/sponsors): By becoming a GitHub sponsor of [jsDelivr](https://github.com/jsdelivr), you are supporting both the jsDelivr CDN and the Globalping platform. Sponsors contributing $10 or more per month can request a hardware probe to install in their home or office.
```
jsDelivr sponsors are unrelated to GP sponsors, lets link here instead to avoid confusion https://github.com/sponsors/jsdelivr

inventarSarah replied:
I replaced the first link -- hope that's what you meant

jimaek replied:
I dont think you did, I dont see any changes.

inventarSarah replied:
Thank you -- now it's updated!

jimaek commented:
I stopped the review at the last comment, will continue after the updates. 
Also we're about to ship a new feature https://github.com/jsdelivr/globalping/pull/453
Lets document it here as well.

I think it should be in the tips and tricks section and maybe a mention in the CLI section. The idea is that the magic field can accept not just locations but also previous measurement IDs to select the exact same probes. This is useful for running comparisons of endpoints from the same probes for better data. Or to run continuous tests from the same probes

inventarSarah commented on README.md:
```diff
@@ -2,54 +2,120 @@
+- [Become a hardware probe provider](https://docs.google.com/document/d/1xIe-BaZ-6mmkjN1yMH5Kauw3FTXADrB79w4pnJ4SLa4/edit#heading=h.tmcgof2zm3yh): Sponsor a batch of co-branded hardware probes, including your own stickers and swag, to ship to your users or hand out at conferences.
+
+## Get started writing tests
+While we offer [different tools and integrations](#available-globalping-tools-and-integrations) for running tests with Globalping, we make sure that they provide a consistent experience. This also means that writing tests looks almost identical whether you're using the CLI tool or the Slack app, for example. 
```
I added a new section before this one: a Quick Start section that shows a simple example of how to use GP with the web tool and the client. I also added a screenshot.

This is basically what I came up with as a compromise between the original readme and my updates -- let me know if that works for you.

So the flow of information is:
- quick start, write your first test (get an idea about Globalping)
- test writing basics (learn the basics of writing tests)
- available integrations (learn what tools you can use to improve your experience)
- tips and best practices (learn advanced techniques and tips to get the most out of GP)

jimaek replied:
Sorry I still prefer the previous version. It has clear links as headers, simple descriptions but they are not too verbose.

jimaek replied:
Also now its duplicating the same content for no good reason

inventarSarah replied:
No worries at all, thanks for your feedback :)

Okay, so, I have moved the integration section up again and renamed it to "quick start ...". I hope this looks good to you
-> I didn't change that much content in these sections; but I removed the repeated info about locations -> I have consolidated this in the "Basics" section. I just think it was very repetitive, but we can add it back if you think we should still mention it for every integration.

After that section we have the "basics of writing tests" and then the tips and best practices. 

So in summary, I removed the newly added section, swapped two sections and updated the headers

inventarSarah commented:
RE. new feature #453 

Updated the content with info about new feature -> still WIP as I need to add examples (added placeholders)

- added info to the "Location" section within "basics of writing tests" 
- added a new section in under "best practices"
- added info to the "no UUIDs" section

inventarSarah commented:
> RE. new feature #453
> 
> Updated the content with info about new feature -> still WIP as I need to add examples (added placeholders)
> 
> * added info to the "Location" section within "basics of writing tests"
> * added a new section in under "best practices"
> * added info to the "no UUIDs" section

I reworked the section with the available information and added examples. 
Once we're happy with the readme, I can move the content for the new feature in a separate PR so you can release it when the feature's available (let me know if we should do this)

jimaek commented on README.md:
```diff
@@ -114,77 +121,77 @@ POST https://api.globalping.io/v1/measurements
 
-### Magic field
-All of our integrations are using the same API and our "magic" field as location input. 
+## How to write networking tests with Globalping
```
I think we need a better title. Again this makes me think of automating some kind of network tests, because thats when you "write" network tests, in this case we "run" them.

Something like "Network measurements and tests"?

jimaek replied:
I just re-read it and I am not sure how useful the whole section is. Maybe we can shorten it? All it says is that you need to check the details of the specific tool that you're using and repeats some basics. e.g. initiate command not needed, location should have a link to the "best practices" section, flags are too specific on the tool and test being used...

jimaek commented on README.md:
```diff
@@ -193,124 +200,167 @@ Other acceptable location examples:
-### Multiple locations
+### Test with "magic" 🧙
+All integrations connect to the same API and use our "magic" field as location input.
+This maintains a consistent and straightforward user experience, allowing you to reuse logic and parameters across all tools.
```
Lets add back the "this is the best way..." to make it clear we recommend using this method when possible

jimaek commented on README.md:
```diff
@@ -193,124 +200,167 @@ Other acceptable location examples:
+This maintains a consistent and straightforward user experience, allowing you to reuse logic and parameters across all tools.
 
-To select multiple locations in a single request you can use a comma. And of course you can combine filters and multiple locations at the same time.
+So, for example, when you run a test from "Germany" using the CLI or Slack app, our API's magic parser processes the location.
```
I think I've seen you mention only the CLI and Slack, but the Web UI uses the same magic field too, same as the bot and all others. Just wanted to make it clear.

jimaek commented on README.md:
```diff
@@ -193,124 +200,167 @@ Other acceptable location examples:
 
-- `from germany, greece, uk, usa, canada, japan, china, south africa --limit 20` will return probes from all of the above countries in roughly equal proportions to fit the limit of 20. 
+### Re-select probes ♻️
+You can also provide the "magic" field with a measurement ID to have the API select the same probes used in a previous measurement. 
```
Lets add something like:

 "This is a best effort action and if some of the probes are no longer online they will be missing from the new results. 
Additionally since measurements expire, after different intervals depending user type, you should not hard-code measurement IDs, as new tests will break after a measurement expires.
Make you sure you consider these limitations in your logic"

jimaek commented on README.md:
```diff
@@ -193,124 +200,167 @@ Other acceptable location examples:
 
-- Cloud regions for Google Cloud and AWS. e.g. `aws-eu-west-1` and `gcp-us-south1` They follow each provider's naming scheme and are prefixed by their name.
-- Eyeball and datacenter networks. e.g. `eyeball-network` and `datacenter-network` Unlike datacenter hosted probes, eyeball network are probes hosted with ISP providers that offer internet access to regular people and small businesses. 
+The `world` location uses a pseudo-random algorithm to select probes while [maintaining certain proportions](https://github.com/jsdelivr/globalping/blob/master/config/default.cjs#L47) per continent.
```
Maybe lets keep the "is special" part? To make it clear that this is the only dynamic location we have, everything else is based on static data

jimaek commented on README.md:
```diff
@@ -193,124 +200,167 @@ Other acceptable location examples:
+For example, when requesting tests `from world --limit 100`, the system aims to return probes proportionally from Africa (5 probes), Asia (15 probes), Europe (30 probes), Oceania (10 probes), North America (30 probes), and South America (10 probes).
 
-Our probes don't expose any unique IDs to target. The recommended way to narrow down your results is to use filters and even stack them together.
+### Level up your location-picking game 📍
```
I think this whole section lost a lot of useful information and became harder to read. I much prefer the previous organization and content. The titles of sections are harder to understand, lots of info was removed, some rewritten sentences lost some explanations and so on. Can we revert this whole section and make smaller changes to it?

jimaek commented on README.md:
```diff
@@ -193,124 +200,167 @@ Other acceptable location examples:
+This whole process is completely automated and managed by our platform.
 
-These limits are applied per IP address regardless if an API key supplied or not.
+## Join the network – Run a virtual probe
```
All of our probes are virtual by default, so lets just call them probes.
It's RIPE Atlas that is doing the opposite, their default is hardware probes, so they would need to use the "virtual" prefix everywhere, but not us.

jimaek commented on README.md:
```diff
@@ -193,124 +200,167 @@ Other acceptable location examples:
 
-- 100 POST requests per minute per IP. No GET limits are implemented to support "real-time" use-cases.
-- A single measurement is limited to 200 probes per location and 500 total probes
+### Quick start guide
```
I don't think this separation is needed, there is no non-quick guide. These instructions are basically all there is to run a probe

jimaek commented on README.md:
```diff
@@ -193,124 +200,167 @@ Other acceptable location examples:
-- 100 POST requests per minute per IP. No GET limits are implemented to support "real-time" use-cases.
-- A single measurement is limited to 200 probes per location and 500 total probes
+### Quick start guide
+You can run a Globalping probe on any machine that can run a Docker container, supporting x86 and ARM architectures. For example, you can use a rented VPS, a dedicated server with spare capacity, or even your locally hosted Raspberry PI.
```
I think you forgot our only requirement "The only requirement is having an internet accessible device able to run docker containers."

jimaek commented on README.md:
```diff
@@ -193,124 +200,167 @@ Other acceptable location examples:
-Anybody can connect to our API and start using it with no credentials required.
-In this case we limit the amount of tests an IP address can run. A single test is defined as a succesful measurement we run and return to the user.
-A limit of 10 tests means the user can run either 10 measurements with the probe limit set to 1 per measurement, or a single measurement with the probe limit set to 10.
+### Notes on probe security and customization
```
Please revert back to the original and make smaller changes if needed. Currently every point has lot some kind of important information. And lets not add a bold header, its not a big focus, people should be able to skim and skip if they want

jimaek commented on README.md:
```diff
@@ -193,124 +200,167 @@ Other acceptable location examples:
+Learn more in the [Globalping Probe respository](https://github.com/jsdelivr/globalping-probe).
 
+## Limits | WIP
+To maintain a sustainable environment, our platform implements several usage limits to prevent abusive behavior while encouraging contributions. Here's an overview:
```
please revert and correct if needed, this sentence changes the point too much

jimaek commented on README.md:
```diff
@@ -114,77 +121,119 @@ POST https://api.globalping.io/v1/measurements
+The location field can process different locations, including continents, regions, countries, cities, US states, and ASNs (prefixed by "AS," e.g., `from AS80085`). You can also specify measurement IDs from previous tests to reuse the same probes.
+>[!TIP]
+>Check out our [best pracises and tips](#basic-location-targeting-) to learn how to define locations effectively. 
+##### Flags
```
Can be removed I think

jimaek commented on README.md:
```diff
@@ -193,124 +242,119 @@ Other acceptable location examples:
 - as396982, as16509
 - comcast, google, ovh, orange, t-mobile
```
Lets add:
- eyeball, datacenter

jimaek commented on README.md:
```diff
@@ -193,124 +242,119 @@ Other acceptable location examples:
 
-When requesting a specific number of probes there is no guarantee that the API will respond back with the exact same amount. Keep this in mind and use it to your benefit.
+#### Probes share no UUIDs
+Our probes don't expose unique IDs that you can use to target them explicitly. Our probes don't expose unique IDs that you can use to target them explicitly. Instead, we recommend fine-tuning the probe selection by using and combining filters or specifying IDs from previous measurements, as described in the best practice section above.
```
duplicate sentence

jimaek commented on README.md:
```diff
@@ -193,124 +242,119 @@ Other acceptable location examples:
+This ensures that popular locations, like `from Amsterdam` or `from AWS`, are automatically load-balanced across multiple probes within the same location, preventing the overload of specific probes.
 
-Globalping only exposes and allows users to interact with online probes. A probe is considered online and becomes available via our API when:
+#### Test result variability
```
I dont think anyone will understand what the title means and will have to read the description. The previous title made it clear immediately what this is about

jimaek commented on README.md:
```diff
@@ -193,124 +242,119 @@ Other acceptable location examples:
-This is completely automated and managed by our platform.
+Here's when a probe is considered online and available through our API:
+- **It's up-to-date.** A probe running an older version will be forced to auto-update before it becomes available.
+- **It passes quality control tests.** Every probe must pass a test, ensuring it's connected to a reasonably stable network without packet loss. This test is done regularly; if a probe fails, it gets taken offline.
```
Lets add:
- It's not a VPN. We block probes that we detect as being hosted behind a VPN or any other proxying technique. Such probes would report incorrect latency and routing data making them unusable.

jimaek commented:
Thank you
